tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py

```diff
@@ -667,20 +668,63 @@ async def with_airbyte_java_connector(context: ConnectorContext, connector_java_
         .with_label("io.airbyte.name", context.metadata["dockerRepository"])
         .with_entrypoint(entrypoint)
     )
+    return await finalize_build(context, connector_container)
 
 
-def with_airbyte_python_connector(context: ConnectorContext, build_platform):
+async def with_airbyte_python_connector(context: ConnectorContext, build_platform: Platform) -> Container:
     pip_cache: CacheVolume = context.dagger_client.cache_volume("pip_cache")
-    return (
+    connector_container = (
         context.dagger_client.container(platform=build_platform)
         .with_mounted_cache("/root/.cache/pip", pip_cache)
         .build(context.get_connector_dir())
         .with_label("io.airbyte.version", context.metadata["dockerImageTag"])
         .with_label("io.airbyte.name", context.metadata["dockerRepository"])
     )
+    return await finalize_build(context, connector_container)
+
+
+async def finalize_build(context: ConnectorContext, connector_container: Container) -> Container:
+    """Finalize build by running finalize_build.sh or finalize_build.py if present in the connector directory."""
+    connector_dir_with_finalize_script = context.get_connector_dir(include=["finalize_build.sh", "finalize_build.py"])
+    finalize_scripts = await connector_dir_with_finalize_script.entries()
+    if not finalize_scripts:
+        return connector_container
+
+    # We don't want finalize scripts to override the entrypoint so we keep it in memory to reset it after finalization
```

Comment:
just curious, why don't we want the scripts to override the entrypoint?

Reply:
I might reconsider it for specific upcoming cases, but I've not seen connector specifically overriding the classic airbyte entrypoint (except your "normalization in container" usecase, which is handled directly in the main code path you contributed to)